ui/jobs/combo_tracker.ts

```diff
@@ -65,146 +93,27 @@ export default class ComboTracker {
-    const comboTracker = new ComboTracker(breakers, callback);
-    // PLD
-    comboTracker.AddCombo([
-      kAbility.FastBlade,
-      kAbility.RiotBlade,
-      kAbility.GoringBlade,
-    ]);
-    // WAR
-    comboTracker.AddCombo([
-      kAbility.HeavySwing,
-      kAbility.Maim,
-      kAbility.StormsEye,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.HeavySwing,
-      kAbility.Maim,
-      kAbility.StormsPath,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Overpower,
-      kAbility.MythrilTempest,
-    ]);
-    // DRK
-    comboTracker.AddCombo([
-      kAbility.HardSlash,
-      kAbility.SyphonStrike,
-      kAbility.Souleater,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Unleash,
-      kAbility.StalwartSoul,
-    ]);
-    // GNB
-    comboTracker.AddCombo([
-      kAbility.KeenEdge,
-      kAbility.BrutalShell,
-      kAbility.SolidBarrel,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.DemonSlice,
-      kAbility.DemonSlaughter,
-    ]);
-    // DRG
-    comboTracker.AddCombo([
-      kAbility.TrueThrust,
-      kAbility.Disembowel,
-      kAbility.ChaosThrust,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.RaidenThrust,
-      kAbility.Disembowel,
-      kAbility.ChaosThrust,
-    ]);
-    // NIN
-    comboTracker.AddCombo([
-      kAbility.SpinningEdge,
-      kAbility.GustSlash,
-      kAbility.AeolianEdge,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.SpinningEdge,
-      kAbility.GustSlash,
-      kAbility.ArmorCrush,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.DeathBlossom,
-      kAbility.HakkeMujinsatsu,
-    ]);
-    // SAM
-    comboTracker.AddCombo([
-      kAbility.Hakaze,
-      kAbility.Jinpu,
-      kAbility.Gekko,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Hakaze,
-      kAbility.Shifu,
-      kAbility.Kasha,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Hakaze,
-      kAbility.Yukikaze,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Fuga,
-      kAbility.Mangetsu,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Fuga,
-      kAbility.Oka,
-    ]);
-    // MCH
-    comboTracker.AddCombo([
-      kAbility.SplitShot,
-      kAbility.SlugShot,
-      kAbility.CleanShot,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.HeatedSplitShot,
-      kAbility.SlugShot,
-      kAbility.CleanShot,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.HeatedSplitShot,
-      kAbility.HeatedSlugShot,
-      kAbility.CleanShot,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.HeatedSplitShot,
-      kAbility.HeatedSlugShot,
-      kAbility.HeatedCleanShot,
-    ]);
-    // DNC
-    comboTracker.AddCombo([
-      kAbility.Cascade,
-      kAbility.Fountain,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Windmill,
-      kAbility.Bladeshower,
-    ]);
+    const comboTracker = new ComboTracker({
+      player: player,
+      comboBreakers: breakers,
+      comboDelayMs: is5x ? 15000 : 30000,
```

Comment:
Is it possible to not duplicate this time between `base.ts` and `combo_tracker.ts`?

Reply:
Yeah, I am just wondering should I put it in constants.ts or just in ComponentManager.

Reply:
I think you could put both values of constants into constants.ts, and then just pick one in each place based on is5x? Maybe there's a better idea, but that's one thought.  I just want to have the numbers in one place.

Reply:
I have the same thought.
But I am thinking "pass is5x to ComboTracker" vs "just pass comboDelayMs that assigned outside to ComboTracker", which is better.

Reply:
Looks good to me, thanks!